streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -784,7 +784,7 @@ public void close() {
     }
 
     @Test
-    public void shouldThrowTaskCorruptedWithoutCheckpointNonEmptyDir() throws IOException {
+    public void shouldThrowTaskCorruptedWithoutPersistentStoreCheckpointAndNonEmptyDir() throws IOException {
         final long checkpointOffset = 10L;
 
         final Map<TopicPartition, Long> offsets = mkMap(
```

Comment:
Not sure if I understand the test: don't we write a checkpoint in L795 for the persistent store?

Reply:
Yeah, I think that's the point of the test: we wrote a checkpoint but it was missing the offset for one of the persistent stores, thus, we should throw a TaskCorruptedException  in `initializeStoreOffsetsFromCheckpoint`

Reply:
Not clear from the context of the code snipped (maybe need to go back to IntelliJ and read the full test setup) that there are multiple partitions. Also wondering why we include `nonPersistentStorePartition` and `irrelevantPartition`? How do we ensure that those don't case the `TaskCorruptedException`?

Reply:
Sorry yeah the relevant part doesn't show up on github. Basically we register
```  
stateMgr.registerStore(persistentStore, persistentStore.stateRestoreCallback);	            
stateMgr.registerStore(persistentStoreTwo, persistentStoreTwo.stateRestoreCallback);	            
stateMgr.registerStore(nonPersistentStore, nonPersistentStore.stateRestoreCallback);
```

but only write the checkpoint for the `persistentStorePartition`, nonPersistentStorePartition` and `irrelevantPartition`. I think the point of the `irrelevantPartition` is to make sure that we detect that the `persistentStoreTwoPartition` offset is missing even though the checkpoint technically has the correct number of offsets in total. ie, that we actually map the offsets to a registered changelog